packages/blade/src/components/Radio/RadioGroup/useRadioGroup.ts

```diff
@@ -0,0 +1,102 @@
+import React from 'react';
+import isUndefined from 'lodash/isUndefined';
+import type { RadioGroupProps } from './RadioGroup';
+import type { RadioGroupContextType } from './RadioContext';
+import { useControllableState } from '~src/hooks/useControllable';
+import { useTheme } from '~components/BladeProvider';
+import { useId } from '~src/hooks/useId';
+
+type UseRadioGroupProps = Pick<
+  RadioGroupProps,
+  | 'isDisabled'
+  | 'labelPosition'
+  | 'validationState'
+  | 'name'
+  | 'value'
+  | 'defaultValue'
+  | 'onChange'
+  | 'neccessityIndicator'
+>;
+
+export type State = {
+  value: string;
+  setValue(value: string): void;
+  removeValue(): void;
+  isChecked(value: string): boolean;
+};
+
+type UseRadioGroupReturn = {
+  state: State;
+  contextValue: RadioGroupContextType;
+  ids: { labelId: string };
+};
+
+const useRadioGroup = ({
+  value,
+  defaultValue,
+  isDisabled,
+  labelPosition,
+  onChange,
+  validationState,
+  neccessityIndicator,
+  name,
+}: UseRadioGroupProps): UseRadioGroupReturn => {
+  const { platform } = useTheme();
+  const idBase = useId('radio-group');
+  const labelId = `${idBase}-label`;
+  const fallbackName = name ?? idBase;
+  const [checkedValue, setValue] = useControllableState({
+    value,
+    defaultValue,
+    onChange: (v: string) => onChange?.({ value: v, name: fallbackName }),
+  });
+
+  // FIX React native bug, where it unchecks the radio on clicking it again
```

Comment:
Is this fixed?

Reply:
Oh yeah I did fixed it, let me check. Maybe merge kerne ke time pe it got messed up.

Reply:
@chaitanyadeorukhkar Yeah I did fixed it. In [here ](https://github.com/razorpay/blade/pull/605/files#diff-82055f61296921ac0c865ece1e8c254f6ebd3bdf0e2dee1c33d2244cfdac1634R95-R96)

Reply:
Remove the comment then?